### DISCUSSION TITLE
How to post the comments on GitHub pull request page using curl.exe?

### DISCUSSION THREAD

**Discussion by ElangoRajendran** [community/community#24304] [Apps, API and Webhooks]
<p>Are there any possibilities to post the comments on Github pull request page using curl.exe?</p>
<p>I have used the below as below mentioned commands, but it’s not working. Could you suggest a solution for this?</p>
<p><strong>curl -s -H “UserName:Token” -X POST -d ‘{“body”: “My Review comments”}’ "<a href="https://api.github.com/repos/UserName/my-docs/issues/11/comments" rel="noopener nofollow ugc">https://api.github.com/repos/UserName/my-docs/issues/11/comments</a>"</strong></p>

**Comment by byrneh** [comment]
<p>Hi <a class="mention" href="/elangorajendran">@ElangoRajendran</a> ,</p>
<p>You did not give a clue as to what error you are getting? Also running curl with debug options rather than -s (silent) can help you when trying to debug errors</p>
<p>I checked myself and was able to successfully POST using REST API using format in Docs<br>
<a href="https://docs.github.com/en/rest/reference/issues#create-an-issue-comment" rel="noopener nofollow ugc">create-an-issue-comment</a></p>
<p>I then compared against what you had and noticed your authentication header record formatting looks wrong.<br>
Using the curl format and syntax you have used the following worked for me, replace your values between {}<br>
curl -H ‘Authorization: Token {your-token-value}’ -X POST -d ‘{“body”: “My Review comments”}’ ‘<a href="https://api.github.com//repos/%7Bowner%7D/%7Brepo%7D/issues/%7Bissue_number%7D/comments" rel="noopener nofollow ugc">https://api.github.com//repos/{owner}/{repo}/issues/{issue_number}/comments</a>’</p>
<p>Give this a try and let us know if this works for you now. Good luck</p>

**Comment by ElangoRajendran** [comment]
<p>Hi <a class="mention" href="/byrneh">@byrneh</a> ,</p>
<p>Thanks for the update. I have tried as you like mentioned but still i have faced below mentioned issue.  Could you suggest to resolve this issue ?</p>
<p><strong>My Command:</strong><br>
curl -H ‘Elango: Token {6XXXXXXXXXa}’ -X POST -d ‘{“body”: “My Review comments”}’ ‘<a href="https://api.github.com//repos/Elango/RepoName/issues/12/comments" rel="noopener nofollow ugc">https://api.github.com//repos/Elango/RepoName/issues/12/comments</a>’</p>
<p><strong>Error details:</strong><br>
curl: (6) Could not resolve host: Token<br>
curl: (3) Failed to convert 6XXXXXXXXXXXa’ to ACE; string contains a disallowed character<br>
curl: (3) unmatched close brace/bracket in URL position 19:<br>
My Review comments}’</p>
<p>Also, if there any possibilities of share here your worked command, Its more helpful for me ?</p>

**Comment by byrneh** [comment]
<aside class="quote no-group" data-username="ElangoRajendran" data-post="3" data-topic="161016">
<div class="title">
<div class="quote-controls"></div>
 ElangoRajendran:</div>
<blockquote>
<p>curl -H ‘Elango: Token {6XXXXXXXXXa}’ -X POST -d ‘{“body”: “My Review comments”}’</p>
</blockquote>
</aside>
<p>Try this<br>
If my token was 1234567890 then<br>
curl -H ‘Authorization: Token 1234567890’ -X POST -d ‘{“body”: “My Review comments”}’</p>

**Comment by ElangoRajendran** [comment]
<p>Hi <a class="mention" href="/byrneh">@byrneh</a>,</p>
<p>Still, I have faced the below issue, I have tried the below-mentioned command.</p>
<p><em>command:</em></p>
<p><strong>curl -H ‘Elango: Token 6XXXXXXXXXa’ -X POST -d ‘{“body”: “My Review comments”}’ ‘<a href="https://api.github.com//repos/Elango/RepoName/issues/12/comments/issues/12/comments" rel="noopener nofollow ugc">https://api.github.com//repos/Elango/RepoName/issues/12/comments/issues/12/comments</a>’</strong></p>
<p><strong>Issue</strong><br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/2/c/2c56f864672812471122b3518aafd77a9435b671.png" data-download-href="/uploads/short-url/6kflbnrsq4ooQvxJkQPdzJsBGPT.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/59197465/181084157-4d418db2-a801-4c34-9fd8-611bba327b1f.png" alt="image" data-base62-sha1="6kflbnrsq4ooQvxJkQPdzJsBGPT" width="690" height="132" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/2/c/2c56f864672812471122b3518aafd77a9435b671_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1294×249 8.84 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I want to post the comment in the pull request comment section. I can able to post the comment manually(directly) but while trying to post the comment using API cant able post, as the above-shared issue facing.</p>
<p>Could you help to resolve the issue?</p>

**Comment by byrneh** [comment]
<p>HI <a class="mention" href="/elangorajendran">@ElangoRajendran</a>, I tested the syntax in Git Bash.  Can you please repeat the command using Git Bash shell command line.<br>
note there is on a single ‘/’ after <a href="http://api.github.com" rel="noopener nofollow ugc">api.github.com</a></p>
<p>If that works, and you still wish to use Windows command prompt it should be  simple matter of adjusting the command syntax for differences in syntax between Bash and Windows command line.<br>
Still feeling hopeful, and have my fingers crossed for you :slight_smile:</p>

**Comment by ElangoRajendran** [comment]
<p>Hi <a class="mention" href="/byrneh">@byrneh</a>,</p>
<p>Thanks for the update, still I have faced this issue?<br>
Could you suggest how to resolve this issue?</p>
<p><strong>Command:</strong></p>
<p><strong>curl -H ‘Elango: Token 6cXXXXXXXXXa’ -X POST -d ‘{“body”: “My Review comments”}’ ‘<a href="https://api.github.com/repos/Elango/RepoName/issues/12/comments" rel="noopener nofollow ugc">https://api.github.com/repos/Elango/RepoName/issues/12/comments</a>’</strong></p>
<p><strong>Error Snap:</strong><br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/c/1/c19249c71a8ed10e426abd69eedbef111aa5f430.png" data-download-href="/uploads/short-url/rCpzvS4H0Xa0suIPk22mxUGYQqk.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/59197465/181084163-9cf39abe-b6a6-40ca-9f47-78268f4cabdc.png" alt="image" data-base62-sha1="rCpzvS4H0Xa0suIPk22mxUGYQqk" width="690" height="224" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/c/1/c19249c71a8ed10e426abd69eedbef111aa5f430_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1310×426 29.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Regards,<br>
Elango R.</p>

**Comment by byrneh** [comment]
<aside class="quote no-group" data-username="byrneh" data-post="2" data-topic="161016">
<div class="title">
<div class="quote-controls"></div>
 byrneh:</div>
<blockquote>
<p>curl -H ‘Authorization: Token {your-token-value}’ -X POST -d ‘{“body”: “My Review comments”}’</p>
</blockquote>
</aside>
<p>You have accidentally replaced text “<strong>Authorization</strong>” with “<strong>Elango</strong>” in the Header it seems, give it retry, and lets hope for success :slight_smile:<br>
curl -H ‘<strong>Authorization</strong>: Token 6cXXXXXXXXXa’ -X POST -d ‘{“body”: “My Review comments”}’</p>

**Comment by ElangoRajendran** [comment]
<p>Hi <a class="mention" href="/byrneh">@byrneh</a> ,</p>
<p>Thanks for your update, it’s working fine :blush:. As using the below-mentioned command,<br>
<strong>curl -H ‘Authorization: Token 6cXXXXXXXXXba’ -X POST -d ‘{ “body”: “My Review comments updated” }’ '<a href="https://api.github.com/repos/Elango/ReoName/issues/12/comments" rel="noopener nofollow ugc">https://api.github.com/repos/Elango/ReoName/issues/12/comments</a>’</strong></p>
<p>While running the above command in Gitbash it’s working fine but while running the same command in the command prompt below mentioned issue occur. Could you suggest resolving this issue?<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/7/5/757b6e45f0cdc0d68b05e532e915c40a4b54af66.png" data-download-href="/uploads/short-url/gLijpZSgWXuCZS9yQRggWaz9foO.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/59197465/181084165-65d9ca6f-33b8-4c01-a835-48786b9e3ff7.png" alt="image" data-base62-sha1="gLijpZSgWXuCZS9yQRggWaz9foO" width="690" height="148" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/7/5/757b6e45f0cdc0d68b05e532e915c40a4b54af66_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1325×285 12.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by byrneh** [comment]
<p>Hi <a class="mention" href="/elangorajendran">@ElangoRajendran</a>, please update your token immediately , it is shown in could not resolve host</p>

**Comment by byrneh** [comment, marked as answer]
<p>Hi <a class="mention" href="/elangorajendran">@ElangoRajendran</a>,</p>
<p>to get the this working on Git CMD line I had to replace the single quotes with double quotes, I also needed to escape the double quotes in the body text. There are probably a few different ways this can be handled, but in In your example here is one that worked for me</p>
<p>curl -H “Authorization: Token 6cXXXXXXXXXba” -X POST -d “{ \“body\”: \“My Review comments updated\” }” “<a href="https://api.github.com/repos/Elango/ReoName/issues/12/comments" rel="noopener nofollow ugc">https://api.github.com/repos/Elango/ReoName/issues/12/comments</a>”.</p>
<p>Please change your token as mentioned above, I have my fingers crossed for a good result for you</p>
<p>If that finally answers/solves your question please mark this post as solved, so it is easy for anyone else looking for the same answer</p>
